.gitignore

```diff
@@ -36,6 +36,7 @@
 /spec/reports/
 /test/tmp/
 /test/version_tmp/
+/translations/*.*
```

Comment:
Why are the translations files being added to the gitignore?

Reply:
I don't remember why, so I'll remove this.

Reply:
I remember now! OK, so the translation directory here is only created/used by my new technique of gathering up inline messages from the source JSX files. This dir is populated on demand by the message generator  - so like if I run the local server. The messages in this dir's translation files will be placed in the normal aggregated translation dir - so in my opinion we should not version the translation files in this directory. The reason I want to version the directory is so that I don't have to (re)create it as part of populating it.

Reply:
I remember now: this new root-level translation dir is just a container to hold the intermediate form of messages generated from inline messages in our JSX (like the ones I introduced in this PR). All of the message placed in this dir will be folded into the normal aggregated_translations directory, so there is no need to keep the intermediate step (the msg files in this dir). 

For the inline messages, they are defined in the JSX and only exist in the root trans dir as a way to be compatible w/ our ancient react-intl translation aggregation process.

Reply:
My concern is around the translation strings getting released with the component. They need to exist in the node module or they won't be found by aggregate translations in the projects that consume it. If the extracted messages are going to the root level instead of the package level then that needs to be changed.

Reply:
Couldn't figure it out and so I moved the translations to the traditional place - in the message files. PR is updated.